libraries/tock-register-interface/src/macros.rs

```diff
@@ -115,7 +116,7 @@ macro_rules! register_bitmasks {
                     match v {
                         $(
                             $(#[$inner])*
-                            x if x == Value::$valname as $valtype => Some(Value::$valname),
+                            _x if _x == $value => Some(Value::$valname),
```

Comment:
Identifiers starting with an underscore are reserved in Rust for unused identifiers (silencing corresponding warnings), which isn't the case here. Would `x_` work?